testsuite/MDAnalysisTests/coordinates/base.py

```diff
@@ -942,61 +926,61 @@ def test_copy(self, func, ts):
         ts = u.trajectory.ts
         func(self, self.name, ts)
 
-    def test_copy_slice(self):
-        for p, v, f in itertools.product([True, False], repeat=3):
-            if not any([p, v, f]):
-                continue
-            ts = self._from_coords(p, v, f)
-            yield self._check_copy, self.name, ts
-            yield self._check_independent, self.name, ts
-            yield self._check_copy_slice_indices, self.name, ts
-            yield self._check_copy_slice_slice, self.name, ts
-
-    def _check_bad_slice(self, p, v, f):
-        ts = self._from_coords(p, v, f)
+    @pytest.fixture(params=filter(any,
```

Comment:
does this `filter` do the same as the other `itertools.product` call above?  If so can we use just one way of doing this (don't care which)

Reply:
```python
filter(any, itertools.product([True, False], repeat=3))
```

is equivalent to

```python
[pvf for pvf in itertools.product([True, False], repeat=3) if any(pvf)]
```

I personally like the filter call here because it is the clearest to read with the indentations due to being a decorator argument. But the problem is that we can't use tuple unpacking like in the list-comprehension above. I found the tuple unpacking more readable in the above example so I left it.

So yeah they are slightly differently written but personally this gives the best readability in each case. I can unify it though if you like. What would you prefer the filter or list-comprehension?

Reply:
Sure, lets use filter then.  It's just confusing to see two ways for this, implies that maybe there's a difference

Reply:
changed.